examples/unix/c11/z_liveliness.c

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+
+#include <signal.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <zenoh-pico.h>
+
+#if Z_FEATURE_LIVELINESS == 1
+
+static volatile int keepRunning = 1;
+
+void intHandler(int dummy) {
```

Comment:
Without this, neither session closing nor token undeclaration is called at all, the example with zenoh-pico is simply interrupted. Yes, after the session loss timeout, the router will report the loss of the token, but for example in multicast peer-to-peer this will not happen.